### PR TITLE
Improvement/2149 documentation access from the ui

### LINKED ISSUE #2149: Documentation Access from the UI
**Component**:

'ui', 'documentation'

**Why this is needed**:

Since we are going to embed html documentation in UI container (#1595), we would like to have a short and easy link from MetalK8s UI to the HTML documentation 

**What should be done**:

- Have an entry in the top bar to access the documentation. When the user click on the link, documentation is opened in a new tab
- Have a short link in the Volume Creation page to directly access the Volume Creation section in the doc. Ideally in the top right corner of the page
- Just next to the storage class combo, it would be great to have a little question mark that opens storage class documentation section in a new tab. The same short link could be available from the banner saying no storage class is available (but I don't see when this case could happen)

**Implementation proposal** (strongly recommended):

**Test plan**:

### PR DESCRIPTION
**Pull request by ChengYanJin**
**Component**: ui, doc

**Context**: 
We would like to access the doc from UI.

**Summary**:
- Add the documentation entry point from the Navbar 
- Add the link in the volume creation page
- Change the icon in clusterMonistering page ( question icon => infomation icon)
- Fix bugs for create volume list 

**Acceptance criteria**: 
Add the link in the volume creation page and add the documentation entry point from the Navbar:

image

- should pass the cypress test:
`./node_modules/cypress/bin/cypress run -b chrome -s cypress/integration/AccessDocumentation.feature`

---

Closes: #2149 

The integration branch in 2.5 is:
 `w/2.5/improvement/2149-documentation-access-from-the-UI`

### CODE REVIEW

File: ui/src/translations/en.json [hunk 1]
```diff
@@ -131,7 +131,10 @@
...
   "value": "Value",
-  "prometheus_not_available":"Prometheus is not available.",
-  "please_refer_to":"Please refer to",
-  "import_solution_hint":"You should import a solution to prepare your environment"
+  "prometheus_not_available": "Prometheus is not available.",
+  "please_refer_to": "Please refer to",
+  "import_solution_hint": "You should import a solution to prepare your environment",
+  "documentation": "Documentation",
+  "storage_class_creation_documentation": "storage class creation documentation",
+  "volume_creation_documentation": "volume creation documentation"
```
Comment (slaperche-scality) [inline ui/src/translations/en.json]: Given that those icon are located next to the item they document, maybe `Access to the documentation`/`Accéder à la documentation` is enough.

File: ui/src/containers/ClusterMonitoring.js [hunk 2]
```diff
@@ -222,7 +222,7 @@ const ClusterMonitoring = props => {
               </TooltipContent>
             }
           >
-            <QuestionMarkIcon className="fas fa-question-circle" />
+            <QuestionMarkIcon className="fas fa-info-circle" />
```
Comment (slaperche-scality) [inline ui/src/containers/ClusterMonitoring.js]: Not related to this PR (and can't be fixed here, I supposed it's coming from core-ui), but the naming seems poor to me: we call `QuestionMarkIcon` something that's not always a question mark icon…

**Review by gdemonet — COMMENTED** [review]
Code LGTM, will test soon :)

File: ui/src/containers/ClusterMonitoring.js (hunk 2 above)
Comment (ChengYanJin) [reply ui/src/containers/ClusterMonitoring.js]: Forgot to change the name `QuestionMarkIcon` to `InformationMarkIcon`, my bad.
Sure, it's not related to this PR, but it's a tiny change in order to keep the consistency for the icons. 
And just FYI, the icon we import directly from the font-awsome library ;)

File: ui/cypress/integration/AccessDocumentation.feature [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Feature: AccessDocumentation
+ Scenario: Access documentation from the Navbar
+     Given I log in
+     Then I access the documentation from the navbar
```
Comment (gdemonet) [inline ui/cypress/integration/AccessDocumentation.feature]: nitpick: `Then`s are more about verifying some fact, not performing an action. Wording could reflect this, by adjusting it slightly to:

```gherkin
    Then I can access the documentation from the Navbar
```

File: ui/cypress/integration/common/accessDocumentation.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+Then('I access the documentation from the navbar', () => {
+  const target_url = Cypress.env('target_url');
+
+  cy.visit(target_url, {
+    onBeforeLoad(win) {
+      // win is the the remote page's window object
+      // `cy.stub()` stubs the window.open method
+      cy.stub(win, 'open', url => {
+        // cannot make a HTTP request to the documentation url because cypress doesn't support the `SameSite`
```
Comment (gdemonet) [inline ui/cypress/integration/common/accessDocumentation.js]: I don't really understand this issue, what's this `SameSite` problem?

File: ui/cypress/integration/common/accessDocumentation.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        // cy.request(url_doc).then(response => {
+        //   expect(response.status).to.eq(200);
+        // });
+        // what we can do for the moment is to check the url
+        expect(url).to.have.string('/index.html');
```
Comment (gdemonet) [inline ui/cypress/integration/common/accessDocumentation.js]: Can we check the full path, and that it includes the `/docs` Ingress path as well?

**Review by gdemonet — COMMENTED** [review]
Really minor comments, waiting for your answer to give an approval, but looks good to me otherwise :)

File: ui/cypress/integration/AccessDocumentation.feature (hunk 3 above)
Comment (ChengYanJin) [reply ui/cypress/integration/AccessDocumentation.feature]: It's definitely a good point to improve!!!

File: ui/cypress/integration/common/accessDocumentation.js (hunk 4 above)
Comment (ChengYanJin) [reply ui/cypress/integration/common/accessDocumentation.js]: As we discussed, we are not supposed to nest the `cy.resquest` inside the `cy.visit`
I just check if the doc URL is correct. 
For the doc endpoint, we will need a pytest to cover it.

**Review by gdemonet — DISMISSED** [review]
LGTM

**Review by gdemonet — APPROVED** [review]

**Pull request merged**